ClinGen/clincoded
# Connects to #780 kl genomic context links

This PR fixes a bug of displaying genomic context links in header. As discussed in ticket #780, a +/- 30 bp length centered at variation point is required and passed in request query.

**Code Changes**
Added codes to get chromosome, start and stop points from HGVS terms GRCh38 and GRCh37 and to generate and display links in file `src/clincoded/static/components/variant_central/record_gene_disease.js`.

**Test**
* Select ClinVar Variation ID, enter 139214, Save. Expect to see links in header as:
![screen shot 2016-07-13 at 10 07 29 pm](https://cloud.githubusercontent.com/assets/9206696/16828629/51241486-4946-11e6-8c5a-718a5090bdf6.png)
* Click link UCSC GRCh38/hg38 and expect to open a new window as below.
![screen shot 2016-07-13 at 10 12 35 pm](https://cloud.githubusercontent.com/assets/9206696/16828688/f8655e80-4946-11e6-9735-35e59fbb9c9e.png)
* Click link UCSC GRCh37/hg19 and expect to open:
![screen shot 2016-07-13 at 10 14 35 pm](https://cloud.githubusercontent.com/assets/9206696/16828712/35ff601a-4947-11e6-92f7-439eb37f4be3.png)
* Click link Variation Viewer GRCh38 and expect to go to:
![screen shot 2016-07-13 at 10 16 34 pm](https://cloud.githubusercontent.com/assets/9206696/16828736/7f284996-4947-11e6-9235-114460585394.png)
* Click link Variation Viewer GRCh37 and expect to go to:
![screen shot 2016-07-13 at 10 17 37 pm](https://cloud.githubusercontent.com/assets/9206696/16828759/a5359f76-4947-11e6-9a16-3db7795e615a.png)
* Click link Ensembl Browser GRCh38 and expect to go to:
![screen shot 2016-07-13 at 10 18 47 pm](https://cloud.githubusercontent.com/assets/9206696/16828784/cc84307e-4947-11e6-86d8-1b900a9efce1.png)
*Click link Ensembl Browser GRCh37 and expect to go to:
![screen shot 2016-07-13 at 10 19 39 pm](https://cloud.githubusercontent.com/assets/9206696/16828796/ec0b8604-4947-11e6-9772-5da68f615c39.png)

* Go back to variant select page, select ClinGen Allele Registry ID , enter CA501058 (variant with GRCh38 NC_ term only) and Save. Expect to see:
![screen shot 2016-07-13 at 10 21 50 pm](https://cloud.githubusercontent.com/assets/9206696/16828834/3c6a5760-4948-11e6-8b6d-2cead2a117f4.png)

* Go back to variant select page, select ClinGen Allele Registry ID, enter CA501064 (variant without GRCh38 and GRCh37 NC_ terms) and Save. Expect not to see any genomic context links.
![screen shot 2016-07-13 at 10 23 43 pm](https://cloud.githubusercontent.com/assets/9206696/16828870/90041e1a-4948-11e6-8d63-f9dc1a02a562.png)

**End Test**
